File: src/main/scala/dxw405/gui/TopPanel.scala
package dxw405.gui

import java.awt.event.ActionListener
import java.awt.{BorderLayout, FlowLayout}
import javax.swing.{JButton, JMenuBar, JPanel}

import dxw405.gui.GUIAction.GUIAction

class TopPanel(buttonListener: ActionListener) extends JMenuBar {

	private var _downloadButton: JButton = null

	def downloadButton = _downloadButton

	{
		setLayout(new BorderLayout())

		val buttons = new JPanel(new FlowLayout(FlowLayout.LEFT))
		_downloadButton = addButton(buttonListener, "Download", GUIAction.DownloadButtonPressed)
		buttons.add(_downloadButton)
		buttons.add(addButton(buttonListener, "Exit", GUIAction.Exit))
		add(buttons, BorderLayout.CENTER)
	}

	private def addButton(buttonListener: ActionListener, label: String, action: GUIAction): JButton = {
		val goButton = new JButton(label)
		goButton.setActionCommand(action.toString)
		goButton.addActionListener(buttonListener)
		goButton
	}
}

File: src/main/java/dxw405/gui/TopPanel.java
package dxw405.gui;

import javax.swing.*;
import java.awt.FlowLayout;
import java.awt.event.ActionListener;

public class TopPanel extends JPanel
{
	public TopPanel(ActionListener buttonListener)
	{
		setLayout(new FlowLayout(FlowLayout.RIGHT));

		JPanel container = new JPanel();
		container.add(addButton(buttonListener, "Download", Action.DOWNLOAD_BUTTON_PRESSED));
		container.add(addButton(buttonListener, "Exit", Action.EXIT));

		add(container);
	}

	private JButton addButton(ActionListener buttonListener, String label, Action action)
	{
		JButton goButton = new JButton(label);
		goButton.setActionCommand(action.toString());
		goButton.addActionListener(buttonListener);
		return goButton;
	}
}
